File: src/Services/PollService/Poll.Domain/Entities/Poll.cs
namespace Poll.Domain.Entities;

public class Poll
{
    public Guid Id { get; set; }

    public string Question { get; set; }
        = string.Empty;

    public DateTime CreatedAt { get; set; }
        = DateTime.UtcNow;

    public ICollection<PollOption> Options { get; set; }
        = new List<PollOption>();

    public Guid? CreatorId { get; set; }

    public int Upvotes { get; set; }

    public int Downvotes { get; set; }
}

File: frontend/src/types/Poll.ts
export interface PollOption {
    id: string;
    optionText: string;
    voteCount: number;
}

export interface Poll {
    id: string;
    question: string;
    createdAt: string;
    creatorId: string | null;
    upvotes: number;
    downvotes: number;
    userVoteOptionId: string | null;
    userReaction: boolean | null;
    options: PollOption[];
}
